cleanup: Move common files to deploy folder

karthik-us opened the pull request:
Few common files related to deployments were kept
in the examples folder initially. Moving them to
deploy folder and updating the relevant files.

nixpanic commented on docs/deploy-rbd.md:
````diff
@@ -134,7 +134,7 @@ for more information.
 
 ```bash
-kubectl create -f ../example/ceph-config.yaml
+kubectl create -f ../../../examples/ceph-conf.yaml
````
Loos good to me, thanks!

I'm just wondering why this file is not included in the `deploy/...` directory. Would it make sense to move/copy it there?

karthik-us replied:
I do not have any idea why it has been kept outside the `deploy/` directory. If there is no solid reason for keeping it in the `examples/` directory and everyone agrees on this change, will move this to `/deploy/` dir, which needs updates in few other files as well. Or I can copy it to `deploy/` dir, so that we can avoid some diffs.

Madhu-1 replied:
it was kept in example folder when we started initially, common files like [service-monitor.yaml](https://github.com/ceph/ceph-csi/blob/devel/examples/service-monitor.yaml), [csi-config-map-sample.yaml](https://github.com/ceph/ceph-csi/blob/devel/examples/csi-config-map-sample.yaml) and [ceph-conf.yaml](https://github.com/ceph/ceph-csi/blob/devel/examples/ceph-conf.yaml) can be moved to deploy folder and the documentation and e2e test cases need to be adjusted for the same.

karthik-us replied:
Thanks @Madhu-1 for the background.
Just to confirm, you are asking to move all those 3 common files under `deploy/` or just the ceph-conf.yaml file?

Madhu-1 replied:
yes as its applicable to both cephfs/nfs and rbd you can move these 3 files to deploy folder

karthik-us replied:
Done.

nixpanic commented on docs/deploy-cephfs.md:
````diff
@@ -148,7 +148,7 @@ for more information.
 
 ```bash
-kubectl create -f ../../../examples/ceph-conf.yaml
+kubectl create -f ../../../deploy/ceph-conf.yaml
````
it is assumed you are in a subdir of the `deploy` directory already, you can remove a few `../` and `deploy/` from the path now.

karthik-us replied:
Argh.. Missed it in the previous update. Corrected now.

nixpanic reviewed (changes requested):
Just a nit for updating/reducing the path in the doc now that you moved the files.

Feel free to squash the commits in a single one too. Thanks for the continued efforts!

karthik-us committed:
cleanup: Move common files to deploy folder

Few common files related to deployments were kept
in the examples folder initially. Moving them to
deploy folder and updating the relevant files.

Signed-off-by: karthik-us <ksubrahm@redhat.com>

karthik-us commented:
@nixpanic & @riya-singhal31 please review.

nixpanic commented:
I think the full e2e jobs need to run, not sure why Mergify added the `ci/skip/e2e` label (and no idea if  it will add it again after I removed it).

nixpanic commented:
This PR can be rebased (by Mergify) and CI jobs started once #3863 is merged.

nixpanic commented:
@mergifyio rebase

nixpanic commented:
/test all

nixpanic commented:
/test ci/centos/k8s-e2e-external-storage

nixpanic commented:
/test ci/centos/mini-e2e-helm

nixpanic commented:
/test ci/centos/mini-e2e

nixpanic commented:
/test ci/centos/upgrade-tests

nixpanic commented:
@mergifyio queue

Make this PR the 1st in the queue.

nixpanic commented:
@Mergifyio requeue

Let's try that again

nixpanic commented:
Both upgrade CI jobs failed with a similar issue:
```
    STEP: checking provisioner deployment is running 06/01/23 10:07:48.476
    Jun  1 10:07:48.479: INFO: deployment status: expected replica count 1 running replica count 0
```

The provisioner just does not get into a running state... I don't immediately see what the problem could be though.

karthik-us commented:
@nixpanic I was checking the code base to see whether I missed updating any part of the code. Nothing seemed suspicious to me.
As I am fairly new to this space, any help from the experts in debugging this is highly appreciated :)

yati1998 commented:
The CI also says,
```
Jun  1 09:27:11.014: INFO: At 2023-06-01 09:17:07 +0000 UTC - event for csi-cephfsplugin-provisioner-64659d6f45-tjsw8: {kubelet minikube} FailedMount: MountVolume.SetUp failed for volume "ceph-csi-config" : configmap "ceph-csi-config" not found

    Jun  1 09:27:11.014: INFO: At 2023-06-01 09:17:07 +0000 UTC - event for csi-cephfsplugin-provisioner-64659d6f45-tjsw8: {kubelet minikube} FailedMount: MountVolume.SetUp failed for volume "ceph-config" : configmap "ceph-config" not found
```

Seems the config file paths are not correctly mentioned, will re-verify this from my end too.

nixpanic commented:
Because the upgrade tests fail, I suspect that the tests need a fallback to the old path of the files. When an older version is deployed by the new/updates e2e suite, the path for the files is incorrect (the old version only has the old paths).

karthik-us commented:
> Because the upgrade tests fail, I suspect that the tests need a fallback to the old path of the files. When an older version is deployed by the new/updates e2e suite, the path for the files is incorrect (the old version only has the old paths).

Thanks @nixpanic. I was also thinking in the same line. Will add a fallback mechanism to the tests to look at the older paths in case the new one(s) doesn't exist.

karthik-us commented:
> The CI also says,
> 
> ```
> Jun  1 09:27:11.014: INFO: At 2023-06-01 09:17:07 +0000 UTC - event for csi-cephfsplugin-provisioner-64659d6f45-tjsw8: {kubelet minikube} FailedMount: MountVolume.SetUp failed for volume "ceph-csi-config" : configmap "ceph-csi-config" not found
> 
>     Jun  1 09:27:11.014: INFO: At 2023-06-01 09:17:07 +0000 UTC - event for csi-cephfsplugin-provisioner-64659d6f45-tjsw8: {kubelet minikube} FailedMount: MountVolume.SetUp failed for volume "ceph-config" : configmap "ceph-config" not found
> ```
> 
> Seems the config file paths are not correctly mentioned, will re-verify this from my end too.

Thanks @yati1998 for looking at it. Let me incorporate the fix suggested by @nixpanic. Hope that should resolve it.

karthik-us commented:
> Because the upgrade tests fail, I suspect that the tests need a fallback to the old path of the files. When an older version is deployed by the new/updates e2e suite, the path for the files is incorrect (the old version only has the old paths).

Added a fallback mechanism to use the old path in case the new one doesn't exist. Please review.

karthik-us commented:
/test ci/centos/upgrade-tests

nixpanic commented on e2e/utils.go:
```diff
@@ -1732,3 +1732,13 @@ func rwopMayFail(err error) bool {
+
+// getConfigFile returns the passed config file location if it exists, else
+// returns the old location of the config file under 'examples/' directory.
+func getConfigFile(configFile string) string {
```
This works, but it could be done cleaner. The function should not be specific for `cephConfconfigMap`, I think.

Could you make it like this:
```go
func getConfigFile(filename, preferred, fallback string) string {
```

If the file does not exist in the `preferred` directory, check the `fallback` directory instead. Maybe even `panic()` if the file can not be found at all?

karthik-us replied:
Yeah this is much cleaner. Will update it.

I think `panic()` is not required in this section as we get the appropriate error in the `Do(action)` part based on the file's existence and `allowMissing` value.

karthik-us replied:
Done.

karthik-us committed:
e2e: Make getConfigFile() generic

Update the getConfigFile() function to allow any file to be looked
at on the preferred location first and fall back to the alternate
location if it does not exist there.

Signed-off-by: karthik-us <ksubrahm@redhat.com>

karthik-us commented:
/test ci/centos/upgrade-tests

nixpanic commented:
@mergifyio rebase

nixpanic commented:
@mergifyio queue

karthik-us commented:
/assign @karthik-us

nixpanic commented:
/retest ci/centos/mini-e2e/k8s-1.25

nixpanic commented:
> /retest ci/centos/mini-e2e/k8s-1.25

Deploying rook seems to have failed? https://jenkins-ceph-csi.apps.ocp.cloud.ci.centos.org/blue/organizations/jenkins/mini-e2e_k8s-1.25/detail/mini-e2e_k8s-1.25/505/pipeline

karthik-us commented:
> Deploying rook seems to have failed? https://jenkins-ceph-csi.apps.ocp.cloud.ci.centos.org/blue/organizations/jenkins/mini-e2e_k8s-1.25/detail/mini-e2e_k8s-1.25/505/pipeline

Is there any problem with the Jenkins instance? Now the rook deployment succeeded, but e2e run seems to be stuck.
https://jenkins-ceph-csi.apps.ocp.cloud.ci.centos.org/blue/organizations/jenkins/mini-e2e_k8s-1.25/detail/mini-e2e_k8s-1.25/506/pipeline/
 How much time does it usually take?

karthik-us commented:
> > Deploying rook seems to have failed? https://jenkins-ceph-csi.apps.ocp.cloud.ci.centos.org/blue/organizations/jenkins/mini-e2e_k8s-1.25/detail/mini-e2e_k8s-1.25/505/pipeline
> 
> Is there any problem with the Jenkins instance? Now the rook deployment succeeded, but e2e run seems to be stuck. https://jenkins-ceph-csi.apps.ocp.cloud.ci.centos.org/blue/organizations/jenkins/mini-e2e_k8s-1.25/detail/mini-e2e_k8s-1.25/506/pipeline/ How much time does it usually take?

Finally it passed. Thanks @nixpanic for the regular follow ups on this and thanks to all the reviewers.
